src/algorithm/neighbour/fastpair.rs

```diff
@@ -0,0 +1,546 @@
+#![allow(non_snake_case)]
+use itertools::Itertools;
+///
+/// FastPair: Data-structure for the dynamic closest-pair problem.
+///
+/// Reference:
+///  Eppstein, David: Fast hierarchical clustering and other applications of
+///  dynamic closest pairs. Journal of Experimental Algorithmics 5 (2000) 1.
+///
+use std::collections::HashMap;
+
+use crate::algorithm::neighbour::dissimilarities::PairwiseDissimilarity;
+use crate::error::{Failed, FailedError};
+use crate::linalg::Matrix;
+use crate::math::distance::euclidian::Euclidian;
+use crate::math::num::RealNumber;
+
+///
+/// FastPair factory function
+///
+#[allow(dead_code)]
+pub fn FastPair<T: RealNumber, M: Matrix<T>>(m: &M) -> Result<_FastPair<'_, T, M>, Failed> {
```

Comment:
Can this be a `fn build(m: &M) -> Result<Self, Failed>` static method in `_FastPair` (maybe other name)?

Reply:
Hi. I refactored `FastPair` to have a constructor and it now works as the other struct in the library.